app/models/issue.rb

```diff
@@ -42,11 +44,13 @@ def show_url
     File.join(project.urlbase, 'issue', sub_id.to_s)
   end
 
-  def sub_id
-    (project.issue_ids.index(id) + 1).to_i
+  def self.find_from_project(project, sub_id)
+    project.issues.find_by_sub_id(sub_id)
   end
 
-  def self.find_from_project(project, sub_id)
-    Issue.find(project.issue_ids[sub_id.to_i - 1])
+  def set_sub_id
+    return if sub_id
+      sub_id = project.issues.count
```

Comment:
Since this is run before saving, we should set it equal to `project.issues.count + 1`